scheduler/src/main/java/gov/nasa/jpl/aerie/scheduler/EquationSolvingAlgorithms.java

```diff
@@ -371,12 +374,25 @@ public boolean isApproximation() {
       if (x_n_double < dxLow_long || x_n_double > dxHigh_long) {
         throw new DivergenceException("Looking for root out of prescribed domain :[" + dxLow + "," + dxHigh + "]");
       }
-      Duration ff_x_nminus1 = ff.valueAt(x0), ff_x_n = ff.valueAt(x_n);
+      var ff_x_nminus1 = ff.valueAt(x0);
+      if (ff_x_nminus1.noShorterThan(Duration.negate(toleranceYLow)) && ff_x_nminus1.noLongerThan(toleranceYHigh)) {
+        return new RootFindingResult<>(x0, ff_x_nminus1.plus(y), null);
+      }
+      var ff_x_n = ff.valueAt(x_n);
+      if (ff_x_n.noShorterThan(Duration.negate(toleranceYLow)) && ff_x_n.noLongerThan(toleranceYHigh)) {
+        return new RootFindingResult<>(x_n, ff_x_n.plus(y), null);
+      }
+      if (x0.isEqualTo(x1)) {
+        //particular : only one start possible, derivative would be 0
+        throw new NoSolutionException();
+      }
+      history.add(Pair.of(x0, ff_x_nminus1));
+      history.add(Pair.of(x1, ff_x_n));
       for (int nbIt = 0; nbIt < maxNbIterations; nbIt++) {
         //(f(xn) - f(xn_m1)) / (xn - xn_m1)
         final double localDerivative =
             (float) (ff_x_n.minus(ff_x_nminus1)).in(Duration.MICROSECONDS) / (x_n_double - x_nminus1_double);
-        if (localDerivative == 0) {
+        if (localDerivative == 0 || Double.isNaN(localDerivative)) {
```

Comment:
🔶 Should we add a new `InfiniteDerivative` exception, or maybe change the name of the `ZeroDerivative` exception to encompass both zero and NaN?

Reply:
I updated its name to `ZeroOrInfiniteDerivativeException`

Reply:
It will not lead to a different resolution but I am open to that.

Reply:
I think it should be provided just for clearer debugging, unless this case is rare?

Reply:
I have added ZeroDerivativeException and InfiniteDerivativeException.